Source/ARTWebSocketTransport.m

```diff
@@ -51,7 +56,19 @@ @implementation ARTWebSocketTransport {
 @synthesize delegate = _delegate;
 @synthesize stateEmitter = _stateEmitter;
 
++ (void)setWebSocketClass:(Class)class {
+    websocketClass = class;
+}
+
 - (instancetype)initWithRest:(ARTRestInternal *)rest options:(ARTClientOptions *)options resumeKey:(NSString *)resumeKey connectionSerial:(NSNumber *)connectionSerial {
+
+    static dispatch_once_t setWebSocketClassOnce;
+    dispatch_once(&setWebSocketClassOnce, ^{
```

Comment:
per my earlier comment, also seems over engineered for purpose.

Reply:
Done at 302d146.